api/src/main/java/io/smallrye/stork/CachingServiceDiscovery.java

```diff
@@ -45,73 +41,26 @@ public CachingServiceDiscovery(ServiceDiscoveryConfig config) {
                     period,
                     e);
         }
+
+        this.lastResults = Collections.emptyList();
+        this.instances = Uni.createFrom().deferred(() -> fetchNewServiceInstances(this.lastResults)
+                .invoke(l -> this.lastResults = l)
+                .onFailure().invoke(this::handleFetchError)
+                .onFailure().recoverWithItem(this.lastResults))
+                .memoize().atLeast(this.refreshPeriod);
```

Comment:
does atLeast literally mean at least? What does it mean in practice?

Reply:
It memoizes for at least the given amount of time, then invalidates.

Reply:
`atLeast` is misleading for me then. But the method is a great fit for what we need :)

Reply:
Would `duringAtLeast` have worked better?

Reply:
I mean, `Uni::memoize()` is still marked as an experimental API so... 🤣

Reply:
I read `at least` for `time ∈ [this, +∞)`.
What about `forPeriod`?

Reply:
@michalszynkiewicz let's move to a Mutiny discussion: https://github.com/smallrye/smallrye-mutiny/discussions/categories/ideas

Reply:
done